benchmarking/run-benchmarking-bookstore

```diff
@@ -0,0 +1,207 @@
+
+lazyUnpackStr=""
+triggerUnpackingStr=""
+
+if [ "${lazyUnpack}" == "true" ]
+then
+  lazyUnpackStr="lazy"
+else
+  lazyUnpackStr="nolazy"
+fi
+
+if [ "${triggerUnpacking}" == "true" ]
+then
+  triggerUnpackingStr="trig"
+else
+  triggerUnpackingStr="notrig"
+fi
+
+output="MarlinMTBenchmarking_Lazy.${lazyUnpackStr}_Trigger.${triggerUnpackingStr}_Sigma.${sigmapercent}.txt"
+
+if [ -f ${output} ]
+then
+  rm ${output}
+fi
+
+echo "Running with following settings:"
+echo "  - n cores: ${cores}"
+echo "  - crunch times: ${crunchtimes}"
+echo "  - crunch sigma percentage: $sigmapercent"
+echo "  - lazy unpacking ? $lazyUnpack"
+echo "  - trigger unpacking ? $triggerUnpacking"
+echo "  - LCIO input file: $inputfile"
+echo "  - Marlin steering file: $steeringfile"
+echo "  - Benchmark output file: $output"
+
+# load the file in the cache for next
+echo "Loading input file into cache: ${inputfile}"
+anajob ${inputfile} >> /dev/null
+
+if [[ "$native" == "true" ]] && [[ $memorylayout != "Shared" ]]; then
+	echo "ERROR copied memory layout is native not possible!"
+	exit 1
+fi
+echo "$memorylayout native: $nativeShared REMOVE! TODO"
+
+echo "Start benchmarking ..."
+for c in ${cores}
+do
+  for cr in ${crunchtimes}
+  do
+		crunchSigma=`echo "(${cr}*${sigmapercent})" | bc`
+		for nh  in ${nhists}
+		do
+			echo "Running MarlinMT with:"
+			echo "  general:"
+			echo "  -- N cores:            ${c}"
+			echo "  -- Lazy unpacking:     ${lazyUnpack}"
+			echo "  -- Trigger unpacking:  ${triggerUnpacking}"
+			echo "  cpu_crunch:"
+			echo "  -- Crunch time:        ${cr}"
+			echo "  -- Crunch sigma:       ${crunchSigma}"
+			echo "  histogram filling:"
+			echo "  -- number histograms:  ${nh}"
+			echo "  -- number bins:        ${nbins}"
+			echo "  -- number fills:       ${nfills}"
+			MarlinMT \
+			${steeringfile} \
+			--constant.TriggerUnpacking="${triggerUnpacking}" \
+			--datasource.LCIOInputFiles="${inputfile}" \
+			--datasource.LazyUnpack="${lazyUnpack}" \
+			--CPUCrunch.CrunchTime=${cr} \
+			--CPUCrunch.CrunchSigma=${crunchSigma} \
+			--bookstore.DefaultMemoryLayout=${memorylayout} \
+			--HistFilling.NBins=${nbins} \
+			--HistFilling.NHists=${nh} \
+			--HistFilling.NFills=${nfills} \
+			--HistFillingNative.NBins=${nbins} \
+			--HistFillingNative.NHists=${nh} \
+			--HistFillingNative.NFills=${nfills} \
+			--constant.RunNative=${native} \
+			--global.Concurrency=${c} > scaling_temp_$$.log 
+			if [ $? -ne 0 ]; then
+				echo "MarlinMT failed!, stop benchmarks!"
+				exit 1
+			fi
+			serial=$( cat scaling_temp_$$.log | grep serial | awk '{print $7}' )
+			parallel=$( cat scaling_temp_$$.log |  grep serial | awk '{print $9}' )
+			scaling=$( cat scaling_temp_$$.log | grep serial| awk '{print $11}' )
+			echo "Result: "
+			echo "  serial Time:   ${serial}"
+			echo "  parallel Time: ${parallel}"
+			echo "  scaling:       ${scaling}"
+			echo "${c} ${cr} ${crunchSigma} ${nh} ${serial} ${parallel} ${scaling}" >> ${output}
```

Comment:
What kind of plots do you to produce out of this benchmark ? The data you want to write on this line may depends on the answer